src/main/java/org/dependencytrack/tasks/scanners/TrivyAnalysisTask.java

```diff
@@ -0,0 +1,486 @@
+        final String requestUrl = "%s/twirp/trivy.cache.v1.Cache/PutBlob".formatted(apiBaseUrl);
+
+         try {
+            HttpUriRequest request = buildRequest(requestUrl, input);
+
+            try (final CloseableHttpResponse response = RETRY.executeCheckedSupplier(() -> HttpClientPool.getClient().execute(request))) {
+                LOGGER.debug("PutBlob response: " + response.getStatusLine().getStatusCode());
+                return (response.getStatusLine().getStatusCode() >= HttpStatus.SC_OK);
+            }
+        } catch (Throwable  ex) {
+            handleRequestException(LOGGER, ex);
+        }
+        return false;
+    }
+
+
+    private TrivyResponse scanBlob(PutRequest input) {
+        final String requestUrl = "%s/twirp/trivy.scanner.v1.Scanner/Scan".formatted(apiBaseUrl);
+
+        ScanRequest scan = new ScanRequest();
+
+        scan.setTarget(input.getDiffID());
+        scan.setArtifactID(input.getDiffID());
+        scan.setBlobIDS(new String[] {input.getDiffID()});
+
+        Options opts = new Options();
+        opts.setVulnType(new String[] {"os", "library"});
+        opts.setScanners(new String[] {"vuln"});
+
+        scan.setOptions(opts);
+
+         try {
+            HttpUriRequest request = buildRequest(requestUrl, scan);
+
+            try (final CloseableHttpResponse response = RETRY.executeCheckedSupplier(() -> HttpClientPool.getClient().execute(request))) {
+
+
+                if (response.getStatusLine().getStatusCode() >= HttpStatus.SC_OK && response.getStatusLine().getStatusCode() < HttpStatus.SC_MULTIPLE_CHOICES) {
+                    String responseString = EntityUtils.toString(response.getEntity());
+                    Gson gson = new Gson();
+                    var trivyResponse = gson.fromJson(responseString,TrivyResponse.class);
+
+                    LOGGER.debug("Scan response: " + response.getStatusLine().getStatusCode());
+                    LOGGER.debug("Response from server: " + responseString);
+                    return trivyResponse;
+                } else {
+                    handleUnexpectedHttpResponse(LOGGER, request.getURI().toString(), response.getStatusLine().getStatusCode(), response.getStatusLine().getReasonPhrase());
+                }
+            }
+        } catch (Throwable  ex) {
+            handleRequestException(LOGGER, ex);
+        }
+        return null;
+    }
+
+    private void deleteBlob(PutRequest input) {
+        final String requestUrl = "%s/twirp/trivy.cache.v1.Cache/DeleteBlobs".formatted(apiBaseUrl);
+        DeleteRequest delete = new DeleteRequest();
+        delete.setBlobIDS(new String[] {input.getDiffID()});
+
+         try {
+            HttpUriRequest request = buildRequest(requestUrl, delete);
+            final CloseableHttpResponse response = RETRY.executeCheckedSupplier(() -> HttpClientPool.getClient().execute(request));
+            LOGGER.debug("DeleteBlob response: " + response.getStatusLine().getStatusCode());
```

Comment:
The response must be closed. At the moment the connection pool gets saturated after ~100 analyses, blocking other requests.

```suggestion
            try (final CloseableHttpResponse response = RETRY.executeCheckedSupplier(() -> HttpClientPool.getClient().execute(request))) {
                LOGGER.debug("DeleteBlob response: " + response.getStatusLine().getStatusCode());
            }
```

Reply:
done